packages/alfa-rules/src/sia-er62/rule.ts

```diff
@@ -38,6 +38,12 @@ const { isText } = Text;
 const { or, not, test } = Predicate;
 const { and } = Refinement;
 
+const hasNonLinkTextCache = Cache.empty<Element, boolean>();
+const pairwiseContrastCache = Cache.empty<
+  Element,
+  Cache<Element, Contrast.Pairing[]>
+>();
+
```

Comment:
This cache should also depend on device (not very important currently), but especially on context (very important for the rule, otherwise we might fetch an `hover` answer when asking for a `focus` pairing or similar problem).
```suggestion
const pairwiseContrastCache = Cache.empty<Device, Cache<Context, Cache<
  Element,
  Cache<Element, ReadonlyArray<Contrast.Pairing>>
>>>();

```
(also, we normally use the `Array<type>` notation rather than `foo[]`, and we likely want that array to be read-only to avoid the risk of changing it after computing it)